service/src/main/kotlin/fi/espoo/evaka/messaging/Message.kt

```diff
@@ -59,16 +58,15 @@ enum class MessageType {
 }
 
 data class MessageReceiversResponse(
-    val groupId: GroupId,
-    val groupName: String,
+    val accountId: MessageAccountId,
     val receivers: List<MessageReceiver>
 )
 
 data class MessageReceiver(
-    val childId: ChildId,
-    val childFirstName: String,
-    val childLastName: String,
-    val childDateOfBirth: LocalDate
+    val id: Id<*>,
+    val type: MessageRecipientType,
+    val name: String,
+    val receivers: List<MessageReceiver>
```

Comment:
Did you try implementing this as `sealed class` with 3 cases instead? I think it would be simpler to understand, plus more type safe because you could use the correct `id` type instead of `Id<*>`.

Furthermore, another commit in this PR seems to add a 4th case...

Reply:
Are there any upsides besides being able to distinguish between the id types? Using sealed classes also means that code generation doesn't work. It just didn't seem like sealed classes bring a lot of value.

Reply:
It's both safer and in my opinion also cleaner. You could model the hierarchy correctly on the type level, so that unit has groups under it and groups have children under it. `Id<*>` allows to put _any_ ID in it, so this doesn't really prevent anything and isn't much better than `UUID`.

I didn't realize that code generation doesn't work 🤔 But adding those types manually to the frontend shouldn't be hard. The correct fix would be to make code generation work instead 😄

Reply:
Modeling the hierarchy doesn't really give anything though since the front end doesn't really care about the hierarchy. This just feels like extra work for no benefit.

Reply:
I had really hard time understanding the diff first, before it occurred to me that this generates the hierarchy. So I think it could make the code cleaner.